Map SO changes and new Map editor

SiogunJH opened the pull request:
Changed Map generation logic, to adjust to design changes.
GridMap now generates map as two GridTile clusters - one for allies and one for enemies.
GridMap editor was modified to account for such drastic changes in GridMap generation logic.

Below is the updated version of GridMap custom editor
![obraz_2025-01-25_170041142](https://github.com/user-attachments/assets/b7c1a838-8e04-4d68-9d4b-c8b58d9b26f0)

SiogunJH committed:
Old GridMap Editor Removal

SiogunJH committed:
GridMap with left and right area

Modified GridMap to generate maps composed of 2 sub-maps (left and right)

SiogunJH committed:
GridManager adjustments

Adjusted GridManager to properly display new GridMap

SiogunJH committed:
GridMap Editor

Added custom editor for GridMap

SiogunJH committed:
Updated SampleGridMap

SiogunJH committed:
SampleGirdMap fix

SiogunJH committed:
Removed warnings

Having GridMap SO simply open in inspector would constantly spam log with failed asserts, so those needed to be removed

SiogunJH committed:
Reverted warning removal

GridMap now validate if its dimension value was changed and if modified object is a runtime instance, before sending an assertion failure

AbexCorp commented on Assets/Scripts/Grid/GridManager.cs:
```diff
@@ -53,52 +53,53 @@ private void DisplayGrid()
+            for (int x = 1; x <= Grid.WidthL; x++)
             {
-                GridTileData tile = Grid[x, z];
+                DisplayTile(x, y);
```
Happens after pressing the "Display Grid" button twice when not in play mode

Occupant cannot be null!
UnityEngine.Debug:Assert (bool,string,UnityEngine.Object)
GridManager:DisplayTile (int,int) (at Assets/Scripts/Grid/GridManager.cs:92)
GridManager:DisplayGrid () (at Assets/Scripts/Grid/GridManager.cs:60)
System.Reflection.MethodBase:Invoke (object,object[])
VInspector.VInspectorData/<>c__DisplayClass0_4:<Setup>b__10 () (at Assets/Addons/vInspector/VInspectorData.cs:78)
VInspector.VIScriptComponentEditor:<OnInspectorGUI>g__drawButtons|0_4 (VInspector.VIScriptComponentEditor/<>c__DisplayClass0_0&) (at Assets/Addons/vInspector/VIScriptComponentEditor.cs:295)
VInspector.VIScriptComponentEditor:OnInspectorGUI () (at Assets/Addons/vInspector/VIScriptComponentEditor.cs:316)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)

MissingReferenceException: The object of type 'GridEntity' has been destroyed but you are still trying to access it.
Your script should either check if it is null or you should not destroy the object.
GridManager.DisplayTile (System.Int32 x, System.Int32 y) (at Assets/Scripts/Grid/GridManager.cs:95)
GridManager.DisplayGrid () (at Assets/Scripts/Grid/GridManager.cs:60)
System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <1fd52158108c4fe3a6817b50854eb75e>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <1fd52158108c4fe3a6817b50854eb75e>:0)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <1fd52158108c4fe3a6817b50854eb75e>:0)
VInspector.VInspectorData+<>c__DisplayClass0_4.<Setup>b__10 () (at Assets/Addons/vInspector/VInspectorData.cs:78)
VInspector.VIScriptComponentEditor.<OnInspectorGUI>g__drawButtons|0_4 (VInspector.VIScriptComponentEditor+<>c__DisplayClass0_0& ) (at Assets/Addons/vInspector/VIScriptComponentEditor.cs:295)
VInspector.VIScriptComponentEditor.OnInspectorGUI () (at Assets/Addons/vInspector/VIScriptComponentEditor.cs:316)
UnityEditor.UIElements.InspectorElement+<>c__DisplayClass76_0.<CreateInspectorElementUsingIMGUI>b__0 () (at <863800b36fd24ebc9a9b0a985ae0ee1a>:0)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr, Boolean&)

SiogunJH committed:
Warning condition rewritten

SiogunJH committed:
DisplayGrid safety check

DisplayGrid was not created with the intention of testing it outside of play mode, so a safety check was added to control the conditions it can run in
